src/ws/Socket.ts

```diff
@@ -341,7 +342,7 @@ export class Socket extends EventEmitter {
      * @fires Socket#error
      */
     public boot() {
-        const ws = this.ws = new this.wsCtor(this.getAddress());
+        const ws = this.ws = new (<WSCtor>this.wsFactory)(this.getAddress());
```

Comment:
eh, I don't _really_ like this since `new`ing arrows will throw an exception that isn't obvious from the type information. But not sure there's a better solution.

Reply:
Can we maybe check if it's a ctor?